### ISSUE TITLE
Build errors in release in terminal\src\tools\closetest\closetest.cpp

### ISSUE THREAD

**Issue by EmJayGee** [microsoft/terminal#12769]
### Windows Terminal version

n/a

### Windows build number

10.0.19044.1620

### Other Software

This s a repo synced to:

C:\Users\micha\source\repos\terminal>git show
commit f5882236dcb8693a0b14c70e4bbe73e6401d844d (HEAD -> main, origin/main, origin/HEAD)
Author: Dan Mezhiborsky <danielmarkm@gmail.com>
Date:   Thu Mar 24 17:08:57 2022 -0400

### Steps to reproduce

Sync to:

C:\Users\micha\source\repos\terminal>git show
commit f5882236dcb8693a0b14c70e4bbe73e6401d844d (HEAD -> main, origin/main, origin/HEAD)
Author: Dan Mezhiborsky <danielmarkm@gmail.com>
Date:   Thu Mar 24 17:08:57 2022 -0400

tools\razzle rel
bcz

...

       "C:\Users\micha\source\repos\terminal\OpenConsole.sln" (Clean;Build target) (1) ->
       "C:\Users\micha\source\repos\terminal\src\host\ft_uia\Host.Tests.UIA.csproj.metaproj" (Clean;Build target) (18)
       ->
       "C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.vcxproj" (default target) (42:3) ->
       (ClCompile target) ->
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(301,17): error C4189: 'count2': local v
       ariable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.vc
       xproj]
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(377,17): error C4189: 'waitRet': local
       variable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.v
       cxproj]
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(486,16): error C4189: 'success': local
       variable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.v
       cxproj]
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(606,20): error C4189: 'success': local
       variable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.v
       cxproj]

    10 Warning(s)
    4 Error(s)

Time Elapsed 00:12:25.35

### Expected Behavior

No warnings, no build errors.

### Actual Behavior

Build FAILED.

       "C:\Users\micha\source\repos\terminal\OpenConsole.sln" (Clean;Build target) (1) ->
       "C:\Users\micha\source\repos\terminal\src\host\ut_lib\host.unittest.vcxproj.metaproj" (Clean;Build target) (9) -
       >
       "C:\Users\micha\source\repos\terminal\src\host\proxy\Host.Proxy.vcxproj" (default target) (15:11) ->
       (Link target) ->
         LINK : warning LNK4266: missing load config symbol for image built with /GUARD [C:\Users\micha\source\repos\te
       rminal\src\host\proxy\Host.Proxy.vcxproj]

       "C:\Users\micha\source\repos\terminal\OpenConsole.sln" (Clean;Build target) (1) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\CascadiaPackage\CascadiaPackage.wapproj.metaproj" (Clean;Buil
       d target) (2) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\CascadiaPackage\CascadiaPackage.wapproj" (default target) (74
       :2) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\WindowsTerminal\WindowsTerminal.vcxproj" (GetTargetPath targe
       t) (76:5) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\Remoting\dll\Microsoft.Terminal.Remoting.vcxproj" (default ta
       rget) (75:5) ->
         LINK : warning LNK4001: no object files specified; libraries used [C:\Users\micha\source\repos\terminal\src\ca
       scadia\Remoting\dll\Microsoft.Terminal.Remoting.vcxproj]

       "C:\Users\micha\source\repos\terminal\OpenConsole.sln" (Clean;Build target) (1) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\CascadiaPackage\CascadiaPackage.wapproj.metaproj" (Clean;Buil
       d target) (2) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\CascadiaPackage\CascadiaPackage.wapproj" (default target) (74
       :2) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\WindowsTerminal\WindowsTerminal.vcxproj" (GetTargetPath targe
       t) (76:5) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsModel\dll\Microsoft.Terminal.Settings.Model.v
       cxproj" (default target) (84:9) ->
         LINK : warning LNK4001: no object files specified; libraries used [C:\Users\micha\source\repos\terminal\src\ca
       scadia\TerminalSettingsModel\dll\Microsoft.Terminal.Settings.Model.vcxproj]

       "C:\Users\micha\source\repos\terminal\OpenConsole.sln" (Clean;Build target) (1) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\CascadiaPackage\CascadiaPackage.wapproj.metaproj" (Clean;Buil
       d target) (2) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\CascadiaPackage\CascadiaPackage.wapproj" (default target) (74
       :2) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\WindowsTerminal\WindowsTerminal.vcxproj" (GetTargetPath targe
       t) (76:5) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Settings.Editor.vcx
       proj" (default target) (87:7) ->
       (MarkupCompilePass2 target) ->
         C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\AddProfile.xaml(63,56): XamlCompiler
       warning WMC1506: OneWay bindings require at least one of their steps to support raising notifications when their
        value changes [C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Sett
       ings.Editor.vcxproj]
         C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\ColorSchemes.xaml(119,44): XamlCompil
       er warning WMC1506: OneWay bindings require at least one of their steps to support raising notifications when th
       eir value changes [C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.S
       ettings.Editor.vcxproj]
         C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Interaction.xaml(58,37): XamlCompiler
        warning WMC1506: OneWay bindings require at least one of their steps to support raising notifications when thei
       r value changes [C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Set
       tings.Editor.vcxproj]
         C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Launch.xaml(107,56): XamlCompiler war
       ning WMC1506: OneWay bindings require at least one of their steps to support raising notifications when their va
       lue changes [C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Setting
       s.Editor.vcxproj]
         C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Launch.xaml(60,56): XamlCompiler warn
       ing WMC1506: OneWay bindings require at least one of their steps to support raising notifications when their val
       ue changes [C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Settings
       .Editor.vcxproj]
         C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Appearances.xaml(56,40): XamlCompiler
        warning WMC1506: OneWay bindings require at least one of their steps to support raising notifications when thei
       r value changes [C:\Users\micha\source\repos\terminal\src\cascadia\TerminalSettingsEditor\Microsoft.Terminal.Set
       tings.Editor.vcxproj]

       "C:\Users\micha\source\repos\terminal\OpenConsole.sln" (Clean;Build target) (1) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\LocalTests_TerminalApp\TerminalApp.LocalTests.vcxproj.metapro
       j" (Clean;Build target) (40) ->
       "C:\Users\micha\source\repos\terminal\src\cascadia\LocalTests_TerminalApp\TerminalApp.LocalTests.vcxproj" (defau
       lt target) (94:2) ->
       (Link target) ->
         TerminalAppLib.lib(XamlTypeInfo.g.obj) : warning LNK4197: export 'DllGetActivationFactory' specified multiple
       times; using first specification [C:\Users\micha\source\repos\terminal\src\cascadia\LocalTests_TerminalApp\Termi
       nalApp.LocalTests.vcxproj]

       "C:\Users\micha\source\repos\terminal\OpenConsole.sln" (Clean;Build target) (1) ->
       "C:\Users\micha\source\repos\terminal\src\host\ft_uia\Host.Tests.UIA.csproj.metaproj" (Clean;Build target) (18)
       ->
       "C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.vcxproj" (default target) (42:3) ->
       (ClCompile target) ->
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(301,17): error C4189: 'count2': local v
       ariable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.vc
       xproj]
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(377,17): error C4189: 'waitRet': local
       variable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.v
       cxproj]
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(486,16): error C4189: 'success': local
       variable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.v
       cxproj]
         C:\Users\micha\source\repos\terminal\src\tools\closetest\closetest.cpp(606,20): error C4189: 'success': local
       variable is initialized but not referenced [C:\Users\micha\source\repos\terminal\src\tools\closetest\CloseTest.v
       cxproj]

    10 Warning(s)
    4 Error(s)

Time Elapsed 00:12:25.35

C:\Users\micha\source\repos\terminal>git show

**Comment by EmJayGee** [comment]
Note that I don't think that that checkin has anything to do with this failure, I was just clarifying where my repro is synchronized to. Or whatever the correct git terminology is.

**Comment by zadjii-msft** [comment]
https://github.com/microsoft/terminal/blob/f5882236dcb8693a0b14c70e4bbe73e6401d844d/src/tools/closetest/closetest.cpp#L301-L302
https://github.com/microsoft/terminal/blob/f5882236dcb8693a0b14c70e4bbe73e6401d844d/src/tools/closetest/closetest.cpp#L377-L378
https://github.com/microsoft/terminal/blob/f5882236dcb8693a0b14c70e4bbe73e6401d844d/src/tools/closetest/closetest.cpp#L486-L488
https://github.com/microsoft/terminal/blob/f5882236dcb8693a0b14c70e4bbe73e6401d844d/src/tools/closetest/closetest.cpp#L606-L608

Huh. So those are used, but they're used in `assert`s, which might be ifdef'd out in release.

But those haven't changed since we released the source code, and they do build and run as part of the CI, so we should have heard about this. 

What version of VS / MSVC are you using?

**Comment by EmJayGee** [comment]
Yeah the issue is definitely that they are only referenced in the asser() macro. I see that some of these were cleaned up in other files using UNREFERENCED_PARAMETER(). I have a CL/PR or whatever the new nomenclature is that fixes it, I'm figuring out what I need to do before sending it in.

I'm using whatever the most recent version of VS is, um, it looks like...

C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.31.31103\bin\HostX64\x64\link.exe

I don't think this is a tooling issue, I think that people aren't building release. At least not using "bcz" / "bz". If these aren't the correct ways to build the project they should either get fixed or removed from the recommendations.

Mike

**Comment by zadjii-msft** [comment]
Just tried a local release build with `bx` locally and yea, this totally repros. 

Sending a PR to just add `UNREFERENCED_PARAMETER` is definitely good enough for me. 


> I don't think this is a tooling issue, I think that people aren't building release. At least not using "bcz" / "bz". If these aren't the correct ways to build the project they should either get fixed or removed from the recommendations.

Well, that's probably not wrong 😛 Usually when I'm building, I'm just using `bx` in the directory I'm actually interested in (since the whole solution is usually a LONG time). This kinda makes sense - these are tests, so usually you only run tests locally in Debug (so you can actually debug why the test failed). 

It's mildly curious that the warnings are ignored in CI instead of reported as errors like they are locally...

**Comment by EmJayGee** [comment]
Assign to me if you please, I have the changes ready. I'm just figuring out how to do the appropriate testing other than just building.

**Comment by EmJayGee** [comment]
I've got this mostly ready, having a hard time getting the unit tests to run (should be no impact, famous last words). Have to figure out Git stuff, not sure if I was supposed to create a branch on github, the text in contributing.md is misworded ("fork" is, as far as I can tell, not defined in git so I don't know if it really meant create a branch at github or if it was just meant to reiterate the "clone the repo locally" part of the workflow. first time for me so sorry for the roadbumps.)

**Comment by floh96** [comment]
@EmJayGee you need to fork the repo clone your fork and then make the changes in your fork and push them to github.
The last step is to open a pr to the terminal repo.
[What is a fork?](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/about-forks)

**Comment by EmJayGee** [comment]
@floh96 thank you! Forks weren't anywhere in the git docs, I didn't think of looking on github. Mind expanded. Thanks again.

**Comment by EmJayGee** [comment]
Sorry this has been languishing, I was figuring out forks and branches and then I got sick over the weekend but I think I'm set up now. I haven't gotten tests to run, they are failing for some reason but they are failing without these changes. I'll take that up as another issue. I'm still sick so I'll nap for now but I'll be back later to check it in or submit it or whatever the new lingo is to the branch I made for the issue then push it to main for my fork and then I guess submit a PR from my fork to the main fork, I think that's how it works.
